server/services/address.service.js

```diff
@@ -56,9 +56,22 @@ module.exports = class AddressService {
   }
 
   static async _queryAddressEndpoint (postcode, pageNumber, pageSize) {
+    /**
+     * Check if the address lookup certificate is a file or a base64 string.
+     * If it's a string convert it back to binary
+    */
+    let addressLookupCert = ''
+    if (config.addressLookupPfxCert) {
+      if (config.addressLookupPfxCert.toUpperCase().endsWith('.PFX')) {
+        addressLookupCert = readFileSync(config.addressLookupPfxCert)
+      } else if (config.addressLookupPfxCert) {
```

Comment:
You don't need the 'if' here because you already have the same statement on line 64.

Once you have got rid of that, you could do the calculation as a ternary to remove the one of the other if's

I.e.
addressLookupCert = config.addressLookupPfxCert.toUpperCase().endsWith('.PFX') ? 
  readFileSync(config.addressLookupPfxCert) :
  Buffer.from(config.addressLookupPfxCert, 'base64')

If this is in a separate helper method (as per the previous comment) then you could just do:

const _getCertificate = () => {

return config.addressLookupPfxCert.toUpperCase().endsWith('.PFX') ? 
  readFileSync(config.addressLookupPfxCert) :
  Buffer.from(config.addressLookupPfxCert, 'base64')

}

Reply:
Thanks Duncan, I'll make those changes

Reply:
I've put it in  a separate helper method as you suggested, looks much more tidy.

You'll notice I wrapped it in `if (config.addressLookupPfxCert)`. This was because without this it was breaking some tests in address.service.test. This appeared to be because the test must have been causing `config.addressLookupPfxCert` to be null in some instances so the `.toUpperCase()` method was getting upset.